pandas/core/frame.py

```diff
@@ -4969,6 +4969,39 @@ def f(x):
 
         return Series(result, index=labels)
 
+    def nunique(self, axis=0, dropna=True):
+        """
+        Return Series with number of distinct observations over requested
+        axis.
+
+        .. versionadded:: 0.20.0
+
+        Parameters
+        ----------
+        axis : {0 or 'index', 1 or 'columns'}, default 0
+            0 or 'index' for row-wise, 1 or 'columns' for column-wise
```

Comment:
actually ``Series.nunique`` is defined in ``pandas.core.base`` (so its the *same* for ``Index``). But these could easily hook into the same doc-string system as I said above.